util/reggen/uvm_reg_base.sv.tpl

```diff
@@ -196,6 +196,11 @@ ${apply_regwen(rb, reg, inst)}\
   % endif
 % endif
 ${make_ral_pkg_window_instances(reg_width, esc_if_name, rb)}
+
+      // Create functional coverage for comportable IP-specific specialized registers.
+      // This function can only be called if it is a root block to get the correct gating condition
+      // and avoid creating duplicated cov.
+      if (this.get_parent() == null && get_en_dv_reg_cov()) create_cov();
```

Comment:
Super nit: you can reference base class' protected members directly in the extended classes

```suggestion
      if (this.get_parent() == null && en_dv_reg_cov) create_cov();
```